stores/pg/src/vector/index.ts

```diff
@@ -205,13 +229,23 @@ export class PgVector extends MastraVector {
 
     const { indexName, dimension, metric = 'cosine', indexConfig = {}, buildIndex = true } = params;
 
+    const indexCacheKey = await this.getIndexCacheKey({ indexName, dimension, type: indexConfig.type, metric });
+    const existingIndexCacheKey = this.createdIndexes.get(indexName);
+    if (existingIndexCacheKey && existingIndexCacheKey === indexCacheKey) {
+      // we already saw this index get created since the process started, no need to recreate it
+      return;
+    }
+    this.createdIndexes.set(indexName, indexCacheKey);
+
     const client = await this.pool.connect();
     try {
       // Validate inputs
       if (!indexName.match(/^[a-zA-Z_][a-zA-Z0-9_]*$/)) {
+        this.createdIndexes.delete(indexName);
         throw new Error('Invalid index name format');
       }
       if (!Number.isInteger(dimension) || dimension <= 0) {
+        this.createdIndexes.delete(indexName);
         throw new Error('Dimension must be a positive integer');
       }
```

Comment:
What happens if we set the index in the map after these two checks? Or even after setupIndex is done?